src/index/CompressedBlockPrefiltering.h

```diff
@@ -127,17 +154,36 @@ class NotExpression : public PrefilterExpression {
   std::unique_ptr<PrefilterExpression> child_;
 
  public:
-  explicit NotExpression(std::unique_ptr<PrefilterExpression> child)
-      : child_(child->logicalComplement()) {}
+  explicit NotExpression(std::unique_ptr<PrefilterExpression> child,
+                         bool makeCopy = false)
+      // If we create a copy, the child expression has already been
+      // complemented while creating the original expression. Thus, just move
+      // the child.
+      : child_(makeCopy ? std::move(child) : child->logicalComplement()) {}
 
   std::unique_ptr<PrefilterExpression> logicalComplement() const override;
+  bool operator==(const PrefilterExpression& other) const override;
+  std::unique_ptr<PrefilterExpression> clone() const override;
+  std::string info(size_t depth) const override;
 
  private:
   std::vector<BlockMetadata> evaluateImpl(
       const std::vector<BlockMetadata>& input,
       size_t evaluationColumn) const override;
 };
 
+//______________________________________________________________________________
+// Instanitate templates with specializations (for linking accessibility)
+template class RelationalExpression<CompOp::LT>;
+template class RelationalExpression<CompOp::LE>;
+template class RelationalExpression<CompOp::GE>;
+template class RelationalExpression<CompOp::GT>;
+template class RelationalExpression<CompOp::EQ>;
+template class RelationalExpression<CompOp::NE>;
+
+template class LogicalExpression<LogicalOperator::AND>;
+template class LogicalExpression<LogicalOperator::OR>;
```

Comment:
I'll try it out the linking problems as soon as I am done with looking at the code.